pkg/apiserver/registry/core/namespacedcloudprofile/strategy.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespacedcloudprofile
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+
+	"github.com/gardener/gardener/pkg/api"
+	"github.com/gardener/gardener/pkg/apis/core"
+	"github.com/gardener/gardener/pkg/apis/core/validation"
+)
+
+type namespacedCloudProfileStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy defines the storage strategy for NamespacedCloudProfiles.
+var Strategy = namespacedCloudProfileStrategy{api.Scheme, names.SimpleNameGenerator}
+
+func (namespacedCloudProfileStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (namespacedCloudProfileStrategy) PrepareForCreate(_ context.Context, obj runtime.Object) {
+	namespacedCloudProfile := obj.(*core.NamespacedCloudProfile)
+
+	dropExpiredVersions(namespacedCloudProfile)
+	namespacedCloudProfile.Status = core.NamespacedCloudProfileStatus{}
+}
```

Comment:
We are not adding `Generation` here, I am also unsure when to add `Generation` and when not to add generation.

/cc @timuthy can you please check?

Reply:
cc @timuthy can you please check this comment?

Reply:
Nice catch! Since there will be a controller which reconciles `NamespacedCloudProfile`s, I think we should follow the best practice and set `Generation` (depending on `.spec`) and `ObservedGeneration`.

Reply:
https://github.com/gardener/gardener/pull/9440/commits/55755b1d44aa693d9e1d71d3512d637add3e1465 adds the`Generation` mechanism. Not sure how `ObservedGeneration` is supposed to be considered there, I could not find a good example. Maybe you can point me in the right direction there.

Reply:
@benedictweis you can check [project reconciler](https://github.com/gardener/gardener/blob/master/pkg/controllermanager/controller/project/project/reconciler.go) or shoot reconciler to get an idea of `ObservedGeneration`.